notifiers/github-app/pkg/controller/status.go

```diff
@@ -0,0 +1,67 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/pod"
+	"knative.dev/pkg/apis"
+)
+
+func (r *GitHubAppReconciler) HandleStatus(ctx context.Context, tr *v1beta1.TaskRun) error {
+	client, err := r.GitHub.NewClient("")
+	if err != nil {
+		return err
+	}
+
+	owner := tr.Annotations[key("owner")]
+	repo := tr.Annotations[key("repo")]
+	commit := tr.Annotations[key("commit")]
+
+	var description *string
+	if m := tr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).GetMessage(); m != "" {
+		description = github.String(m)
+	}
+
+	status := &github.RepoStatus{
+		State:       state(tr.Status),
+		Description: description,
+		TargetURL:   github.String(dashboardURL(tr)),
+		Context:     github.String(tr.GetName()),
+	}
+	_, _, err = client.Repositories.CreateStatus(ctx, owner, repo, commit, status)
+	return err
+}
+
+func dashboardURL(tr *v1beta1.TaskRun) string {
+	// TODO: generalize host, object type.
```

Comment:
We could set an env var like DASHBOARD_HOST that this can function can use.

Reply:
Yup! That's the plan, but holding off for another PR for the time being.